hnn_core/network.py

```diff
@@ -129,6 +118,21 @@ def build(self):
 
         print('Building the NEURON model')
         from neuron import h
+        from .parallel import create_parallel_context
+        from .parallel import clear_last_network_objects
+
+        # make sure ParallelContext has been created (needed for joblibs)
+        create_parallel_context()
+
+        clear_last_network_objects(self)
```

Comment:
I would always clear the network object using context manager so you avoid having to do this. Or am I missing something?

Reply:
I'll try to explain... Network() only gets called in the example file, but net.build() gets called for each trial. Right now clear_last_network_objects() needs to be called for each net.build().

This made me think... what about a NeuronNetwork() object with its own context manager? I think that would clean up this code.

Reply:
_neuron_network class is implemented, but there's still the need to clear objects in `build`